### PR TITLE
feat: Added the github actions to send the slack notifications

### PR DESCRIPTION
**Pull request by ravali-rimmalapudi**
# Fixes #
Closes: [DII-131](https://issues.corp.twilio.com/browse/DII-131)
Dependent PRs: [CLI](https://github.com/twilio/twilio-cli/pull/304), [Homebrew](https://github.com/twilio/homebrew-brew/pull/14)

### Checklist
- [x] I acknowledge that all my contributions will be made under the project's license
- [ ] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [ ] I have read the [Contribution Guidelines](https://github.com/twilio/twilio-cli-core/blob/main/CONTRIBUTING.md) and my PR follows them
- [ ] I have titled the PR appropriately
- [ ] I have updated my branch with the main branch
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary documentation about the functionality in the appropriate .md file
- [ ] I have added inline documentation to the code I modified

If you have questions, please file a [support ticket](https://twilio.com/help/contact), or create a GitHub Issue in this repository.

### CODE REVIEW

**Review by shamantraghav — APPROVED** [review]

**Review by shamantraghav — APPROVED** [review]

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -12,6 +12,21 @@ on:
...
 jobs:
+  notify-start:
+    name: Notify Release Started
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Slack Notification
+        uses: rtCamp/action-slack-notify@v2
+        env:
+          SLACK_WEBHOOK: ${{ secrets.SLACK_WEB_HOOK }}
```
[the remark this answers is not in the document]
Comment (ravali-rimmalapudi) [reply .github/workflows/release.yml:23]: It's a webhook for the #help-dev-interfaces channel.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -104,3 +119,20 @@ jobs:
           BRANCH_NAME: ${{github.event.inputs.cli-branch}}
           REPO_NAME: twilio/twilio-cli
           INPUTS: '{ "change-log": ${{ toJSON(needs.update-api-specs.outputs.change-log) }}, "version-type": "${{needs.update-api-specs.outputs.version-type}}", "homebrew-branch": "${{github.event.inputs.homebrew-branch}}", "homebrew-prerelease": "${{github.event.inputs.homebrew-prerelease}}" }'
+  notify-complete-fail:
+    if: ${{ failure() || cancelled() }}
+    needs: [ update-api-specs, release, triggerCliWorkflow ]
```
[the remark this answers is not in the document]
Comment (ravali-rimmalapudi) [reply .github/workflows/release.yml]: Agree. Updated.

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -104,3 +119,20 @@ jobs:
...
+    name: Notify Release Failed
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Slack Notification
+        uses: rtCamp/action-slack-notify@v2
+        env:
+          SLACK_WEBHOOK: ${{ secrets.ALERT_SLACK_WEB_HOOK }}
```
[the remark this answers is not in the document]
Comment (ravali-rimmalapudi) [reply .github/workflows/release.yml:132]: When the released started or completed, we are sending notifications to #help-dev-interfaces channel, but when release failed we are planning to send it to #alert-dev-interfaces.

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -12,6 +12,21 @@ on:
...
+    steps:
+      - uses: actions/checkout@v2
+      - name: Slack Notification
+        uses: rtCamp/action-slack-notify@v2
+        env:
+          SLACK_WEBHOOK: ${{ secrets.SLACK_WEB_HOOK }}
+          SLACK_COLOR: "#36a64f"
+          SLACK_USERNAME: CLI Release Bot
+          SLACK_ICON_EMOJI: ":ship:"
+          SLACK_TITLE: "Twilio Cli"
```
Comment (kridai) [inline .github/workflows/release.yml:27]: I see the SLACK_TITLE for ` Notify Release Failed` being `Twilio CLI-Core` shouldn't this be the same or specific to twilio cli core?

**Review by kridai — APPROVED** [review]

**Review by shrutiburman — APPROVED** [review]
LGTM!

**Pull request merged**